src/main/java/com/github/manolo8/darkbot/utils/login/LoginUtils.java

```diff
@@ -47,10 +50,69 @@ public static LoginData performUserLogin() {
         Popups.showMessageSync("Login", pane, panel::setDialog);
 
         LoginData loginData = panel.getResult();
-        if (loginData.getPreloaderUrl() == null || loginData.getParams() == null) System.exit(0);
+        if (loginData.getPreloaderUrl() == null || loginData.getParams() == null) {
+            System.out.println("Closed login panel, exited without logging in");
+            System.exit(0);
+        }
         return loginData;
     }
 
+    public static LoginData performAutoLogin(StartupParams params) {
+        String username = params.get(StartupParams.PropertyKey.USERNAME);
+        String password = params.get(StartupParams.PropertyKey.PASSWORD);
+        char[] masterPw = params.getMasterPassword();
+
+        if (username == null || (masterPw == null && (password == null || password.isEmpty()))) {
+            System.err.println("Credentials file requires username and either a password or a master password");
+            System.exit(-1);
+        }
+
+        password = getPassword(username, password, masterPw);
```

Comment:
```suggestion
        if (username != null && (password == null || password.isEmpty)) {
            char[] masterPw = params.getMasterPassword();
            password = getPassword(user, masterPw);
            if (password == null)
                System.err.println("Password for user couldn't be retreived. Check that the user exists and master password is correct.");
        }

        if (username == null || password == null || password.isEmpty()) {
            System.err.println("Credentials file requires username and either a password or a master password");
            System.exit(-1);
        }
```

This way we keep a single exit that will run if a password wasn't provided, and none could be obtained (by using the master pw)